sdk/identity/azure-identity/src/main/java/com/azure/identity/SharedTokenCacheCredential.java

```diff
@@ -86,22 +85,18 @@
 
                 if (accounts.size() == 0) {
                     if (username == null) {
-                        return Mono.error(new RuntimeException("No accounts were discovered in the shared token cache."
-                            + " To fix, authenticate through tooling supporting azure developer sign on."));
+                        return Mono.error(new RuntimeException("SharedTokenCacheCredential authentication unavailable. No accounts were discovered in the cache."));
                     } else {
-                        return Mono.error(new RuntimeException(String.format("User account '%s' was not found in the "
-                            + "shared token cache. Discovered Accounts: [ '%s' ]", username, accounts.values().stream()
-                            .map(IAccount::username).collect(Collectors.joining(", ")))));
+                        return Mono.error(new RuntimeException(String.format("SharedTokenCacheCredential authentication unavailable. No account "
+                            + "matching the specified %s was found in the cache.", username)));
```

Comment:
Same here, we should keep the instructions for the user as to how they'll resolve/fix the issue.